File: ios/GreenSquare.m
#import "GreenSquare.h"
#import "ReactNativeComponent.h"
#import "ReactNativeComponentFactory.h"

@implementation GreenSquare

@synthesize text = _text;

-(instancetype)initWithFrame:(CGRect)frame andText:(NSString *)text {
    _text = text;
    return [super initWithFrame:frame];
}

-(NSDictionary*)getProps {
    NSMutableDictionary *propsMap = [[super getProps] mutableCopy];
    
    if (_text != nil) {
        propsMap[@"text"] = _text;
    }

    return propsMap;
}

-(void)addReactSubview {
    self.reactComponent = [ReactNativeComponentFactory create:GreenSquareReactComponent withInitialProperties:[self getProps]];
     
     [super initializeReactSubview: self.reactComponent];
}

@end

File: android/src/main/java/com/callstack/reactnativebrownfield/GreenSquare.kt
package com.callstack.reactnativebrownfield

import android.content.Context

class GreenSquare
    @JvmOverloads constructor (
        context: Context,
        var text: String?
    ): UIComponent(context) {
    init {
        addReactSubview()
    }

    override fun getProps(): HashMap<String, Any> {
        val map = super.getProps()

        if (text != null) {
            map["text"] = text!!
        }

        return map
    }

    override fun addReactSubview() {
        reactComponent = ReactNativeComponentFactory.create(
            ComponentTypes.GreenSquare,
            context,
            getProps()
        )
        initializeReactSubview()
    }
}
